### PR TITLE
[WIP #3844 #3767] update std.c and std.os.linux functions to use null-terminated pointer types

### LINKED ISSUE #3884: update std.c functions to use null-terminated pointer types
https://github.com/ziglang/zig/blob/9561e7c6b9fb2d9ebcbfd611196db698372ae7bd/lib/std/c.zig#L59-L103

### PR DESCRIPTION
**Pull request by raulgrell**
Shoutout to @stratact for the first attempt!

I'm fairly confident the signatures are correct. Are there any tests that need to be created?

Closes #3884

### CODE REVIEW

File: lib/std/c.zig [hunk 1]
```diff
@@ -71,49 +69,49 @@ pub extern "c" fn close(fd: fd_t) c_int;
...
-pub extern "c" fn mkdir(path: [*]const u8, mode: c_uint) c_int;
-pub extern "c" fn symlink(existing: [*]const u8, new: [*]const u8) c_int;
-pub extern "c" fn rename(old: [*]const u8, new: [*]const u8) c_int;
-pub extern "c" fn chdir(path: [*]const u8) c_int;
-pub extern "c" fn execve(path: [*]const u8, argv: [*]const ?[*]const u8, envp: [*]const ?[*]const u8) c_int;
+pub extern "c" fn mkdir(path: [*:0]const u8, mode: c_uint) c_int;
+pub extern "c" fn symlink(existing: [*:0]const u8, new: [*:0]const u8) c_int;
+pub extern "c" fn rename(old: [*:0]const u8, new: [*:0]const u8) c_int;
+pub extern "c" fn chdir(path: [*:0]const u8) c_int;
+pub extern "c" fn execve(path: [*:0]const u8, argv: [*:0]const ?[*]const u8, envp: [*:0]const ?[*]const u8) c_int;
```
Comment (raulgrell) [inline lib/std/c.zig]: Should this be:
pub extern "c" fn execve(path: [*:0]const u8, argv: [*:0]const ?[*]const u8, envp: [*:0]const ?[*]const u8) c_int;
or
pub extern "c" fn execve(path: [*:0]const u8, argv: [*:0]const [*:0]const u8, envp: [*:0]const [*:0]const u8) c_int;

Comment (tiehuis) [reply lib/std/c.zig]: I think you want something like this:

```zig
pub extern "c" fn execve(path: [:0]const u8, argv: [*:null]const ?[*:0]const u8, envp: [*:null]const ?[*:0]const u8) c_int
```

That is, an "array" of null-terminated strings, where the last string is always null.

**Comment by andrewrk** [comment]
You might have to fix some callsites, which may involve some `@ptrCast` until  #3770 is implemented.

**Comment by raulgrell** [comment]
I'm going through the corresponding function in std.os.linux - I'll make sure it's all updated.

It should eventually close #3767

**Pull request closed without merging**